packages/nx-monorepo/scripts/pnpm/link-bundled-transitive-deps.ts

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env ts-node
+/*! Copyright [Amazon.com](http://amazon.com/), Inc. or its affiliates. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0 */
+import * as path from "node:path";
+import * as fs from "fs-extra";
+import { Lockfile, ResolvedDependencies } from "@pnpm/lockfile-types";
+import * as SemVer from "semver";
+
+function resolvePnpmPath(workspaceDir: string, name: string, version: string) {
+  return path.join(workspaceDir, "node_modules", ".pnpm", `${name.replace("/", "+") }@${version}`, "node_modules", name);
+}
+
+async function linkBundledTransitiveDeps(workspaceDir: string, pkgFolder: string, lockfile?: Lockfile) {
+  const pkgDir = path.join(workspaceDir, pkgFolder);
+  const pkgJson = require(path.join(pkgDir, "package.json"));
+  const bundledDeps: string[] = pkgJson.bundledDependencies || [];
+  if (bundledDeps.length < 1) {
+    // No bundled deps
+    return;
+  }
+
+  if (lockfile == null) {
+    require("@pnpm/logger").default = () => {};
+    lockfile = await (require("@pnpm/lockfile-file")).readWantedLockfile(workspaceDir, { ignoreIncompatible: true });
+
+    if (lockfile == null) {
+      throw new Error("Failed to read lockfile")
+    }
+  }
+
+  const pkgDeps = lockfile.importers[pkgFolder].dependencies!;
+
+  const transitiveDeps: Record<string, { version: string, path: string }> = {};
+
+  function visit(_deps?: ResolvedDependencies) {
```

Comment:
Why is this optional? If it needs to remain optional then we should change line 36 to: if `(!deps || _deps == null)`

Reply:
The conditional to check if undefined, null, or empty is inside the function, rather than have to check before each call.